tests/PHPUnit/Integration/ArchiveTest.php

```diff
@@ -101,6 +114,125 @@ public function test_ArchiveIsLaunched_WhenForceOnBrowserRequest_IsFalse_AndArch
+        $this->assertArchiveBlob($dataArrays, '2013-01-03', array('Actions_Actionsurl' => 'test03', 'Actions_Actionsurl_1' => 'subtable1', 'Actions_Actionsurl_2' => 'subtable2', 'Actions_Actionsurl_5' => 'subtable5'));
+        $this->assertArchiveBlob($dataArrays, '2013-01-04', array('Actions_Actionsurl' => 'test04', 'Actions_Actionsurl_5' => 'subtable45', 'Actions_Actionsurl_6' => 'subtable6'));
+        $this->assertArchiveBlob($dataArrays, '2013-01-05', array('Actions_Actionsurl' => 0));
+        $this->assertArchiveBlob($dataArrays, '2013-01-06', array('Actions_Actionsurl' => 'test06'));
+    }
+
+    public function test_ArchiveBlob_ShouldBeAbleToLoadDifferentArchives_NoMatterWhetherTheyAreStoredSeparatelyOrInACombinedSubtableEntry()
+    {
+        $this->createManyDifferentArchiveBlobs();
+
+        $archive = $this->getArchive('day', '2013-01-01,2013-01-06');
+        $dataArrays = $archive->get(array('Actions_Actionsurl', 'Actions_Actions'), 'blob', 2);
+
+        $this->assertArchiveBlob($dataArrays, '2013-01-01', array('Actions_Actionsurl_2' => 0, 'Actions_Actions_2' => 0));
+        $this->assertArchiveBlob($dataArrays, '2013-01-02', array('Actions_Actions_2' => 'actionsSubtable2', 'Actions_Actionsurl_2' => 'test2'));
+        $this->assertArchiveBlob($dataArrays, '2013-01-03', array('Actions_Actions_2' => 'actionsTest2', 'Actions_Actionsurl_2' => 'subtable2'));
+        $this->assertArchiveBlob($dataArrays, '2013-01-04', array('Actions_Actionsurl_2' => 0, 'Actions_Actions_2' => 0));
+        $this->assertArchiveBlob($dataArrays, '2013-01-05', array('Actions_Actionsurl_2' => 0, 'Actions_Actions_2' => 0));
+        $this->assertArchiveBlob($dataArrays, '2013-01-06', array('Actions_Actionsurl_2' => 0, 'Actions_Actions_2' => 0));
+    }
+
+    private function createManyDifferentArchiveBlobs()
+    {
+        $this->createArchiveBlobEntry('2013-01-01', array('Actions_Actionsurl' => array(0 => 'test01')));
+        $this->createArchiveBlobEntry('2013-01-02', array(
+            'Actions_Actionsurl' => array(0 => 'test02', 1 => 'test1', 2 => 'test2'),
+            'Actions_Actions' => array(0 => 'actions_02', 'subtables' => serialize(array(1 => 'actionsSubtable1', 2 => 'actionsSubtable2', 5 => 'actionsSubtable5')))
+        ));
+        $this->createArchiveBlobEntry('2013-01-03', array(
+            'Actions_Actionsurl' => array(0 => 'test03', 'subtables' => serialize(array(1 => 'subtable1', 2 => 'subtable2', 5 => 'subtable5'))),
+            'Actions_Actions' => array(0 => 'actions_03', 1 => 'actionsTest1', 2 => 'actionsTest2'))
+        );
+        $this->createArchiveBlobEntry('2013-01-04', array('Actions_Actionsurl' => array(0 => 'test04', 5 => 'subtable45', 6 => 'subtable6')));
+        $this->createArchiveBlobEntry('2013-01-06', array('Actions_Actionsurl' => array(0 => 'test06', 'subtables' => serialize(array()))));
+    }
+
+    private function assertArchiveBlob(PiwikArchive\DataCollection $dataCollection, $date, $expectedBlob)
```

Comment:
It might increase clarity to append a verb to this method so we know what it does, eg, `assertArchiveBlobEquals` or `assertArchiveBlobHas`.